According to a tester, commit:
https://cgit.freedesktop.org/gstreamer/gst-plugins-base/commit/?id=9e004b56fe36ec08e77ca3dbb07effac1074e952
causes an enigma2 crash during rewind.

Not tested by myself.

This reverts commit 9e004b56fe36ec08e77ca3dbb07effac1074e952.
---
 gst/playback/gsturidecodebin.c | 13 ++-----------
 1 file changed, 2 insertions(+), 11 deletions(-)

diff --git a/gst/playback/gsturidecodebin.c b/gst/playback/gsturidecodebin.c
index f19f69a23..3d8b1375e 100644
--- a/gst/playback/gsturidecodebin.c
+++ b/gst/playback/gsturidecodebin.c
@@ -2076,10 +2076,8 @@ remove_source (GstURIDecodeBin * bin)
       g_signal_handler_disconnect (source, bin->src_nmp_sig_id);
       bin->src_nmp_sig_id = 0;
     }
-    GST_OBJECT_LOCK (bin);
-    bin->source = NULL;
-    GST_OBJECT_UNLOCK (bin);
     gst_bin_remove (GST_BIN_CAST (bin), source);
+    bin->source = NULL;
   }
   if (bin->queue) {
     GST_DEBUG_OBJECT (bin, "removing old queue element");
@@ -2182,7 +2180,6 @@ static gboolean
 setup_source (GstURIDecodeBin * decoder)
 {
   gboolean is_raw, have_out, is_dynamic;
-  GstElement *source;
 
   GST_DEBUG_OBJECT (decoder, "setup source");
 
@@ -2192,14 +2189,8 @@ setup_source (GstURIDecodeBin * decoder)
   decoder->pending = 0;
 
   /* create and configure an element that can handle the uri */
-  source = gen_source_element (decoder);
-  GST_OBJECT_LOCK (decoder);
-  if (!(decoder->source = source)) {
-    GST_OBJECT_UNLOCK (decoder);
-
+  if (!(decoder->source = gen_source_element (decoder)))
     goto no_source;
-  }
-  GST_OBJECT_UNLOCK (decoder);
 
   /* state will be merged later - if file is not found, error will be
    * handled by the application right after. */
-- 
2.20.1

